common/utilities/time/waiting-on.h

```diff
@@ -0,0 +1,134 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2021 Intel Corporation. All Rights Reserved.
+
+#pragma once
+
+#include <mutex>
+#include <condition_variable>
+#include <thread>
+#include <atomic>
+
+
+namespace utilities {
+namespace time {
+
+// Helper class -- encapsulate a variable of type T that we want to wait on: another thread will set
+// it and signal when we can continue...
+//
+template< class T >
+class waiting_on

```

Comment:
The class seem to mimic std::future, why is it needed?

Reply:
Because we're not using std::future... and the whole mechanism of the dispatcher is written without it.
Perhaps we can refactor this class to use it later, but for now... no.